### DISCUSSION TITLE
7.4.2 vs newer versions processing time

### DISCUSSION THREAD

**Discussion by LeonidR2** [dfinke/ImportExcel#1279] [General]
I was tasked to import Excel file with 3 columns and about 100,000 records.  

$FileContentXLS = Import-Excel -Path $FileName -WorksheetName $WorkSheetName

Version 7.4.2 performs task in 20 seconds, while every new version completes task in 15 minutes. 

Would anyone know reason for such discrepancy? Can same optimization be applied to newer versions?

**Comment by dfinke** [comment]
There have been updates to the Import-Excel function. Do you know what version it started to take 15 minutes?

**Reply by LeonidR2** [reply]
Thank you for responding. 7.5.0 is when slowness started. I tested in each version released since 7.4.2.

**Comment by dfinke** [comment]
Quick look. I'm guessing this PR https://github.com/dfinke/ImportExcel/pull/1161 I did to "improve" things caused a performance problem.

**Reply by LeonidR2** [reply]
Hello, would you be able to let me know if slowness issue will be resolved in the near future? 

Thank you, we truly appreciate your work!

**Comment by dfinke** [comment]
Not soon. I've created this as an issue and will lock the discussion. https://github.com/dfinke/ImportExcel/issues/1299
